File: solutions/_905_sort_array_by_parity/solution.java
package _905_sort_array_by_parity;

class Solution {
    public int[] sortArrayByParity(int[] nums) {
        int L = 0;
        for (int R = 0; R < nums.length; R++) {
            if (nums[R] % 2 == 0) {
                int temp = nums[L];
                nums[L] = nums[R];
                nums[R] = temp;
                L++;
            }
        }
        return nums;
    }
}

/**
 * https://leetcode.com/problems/sort-array-by-parity/
 */

File: solutions/_905_sort_array_by_parity/solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParity = function (nums) {
  let L = 0;
  for (let R = 0; R < nums.length; R++) {
    if (nums[R] % 2 === 0) {
      [nums[L], nums[R]] = [nums[R], nums[L]];
      L++;
    }
  }
  return nums;
};

module.exports = { sortArrayByParity };

/*
https://leetcode.com/problems/sort-array-by-parity/
*/
